cmd/global-query/pkg/distributed/querier.go

```diff
@@ -0,0 +1,74 @@
+package distributed
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/els0r/goProbe/pkg/api/goprobe/client"
+	"github.com/els0r/goProbe/pkg/query"
+	"github.com/els0r/goProbe/pkg/results"
+	"gopkg.in/yaml.v3"
+)
+
+type QuerierType string
+
+const (
+	APIClientQuerierType QuerierType = "api"
+)
+
+type Querier interface {
+	CreateQueryWorkload(ctx context.Context, host string, args *query.Args) (*QueryWorkload, error)
+}
+
+type APIClientQuerier struct {
+	apiEndpoints map[string]*client.Config
+}
+
+func NewAPIClientQuerier(cfgPath string) (*APIClientQuerier, error) {
+	a := &APIClientQuerier{
+		apiEndpoints: make(map[string]*client.Config),
+	}
+
+	// read in the endpoints config
+	f, err := os.Open(cfgPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open config: %w", err)
+	}
+	defer f.Close()
```

Comment:
It's something, I've wondered about. How do you do it?

```
defer func() {
  return f.Close()
}
```
And then what?

Reply:
Assuming there is already an error in the function (like here) I have adopted the following scheme (for functions that are not performance-sensitive, since `defer` is never cheap):
```
defer func() {
	if cerr := metadataFile.Close(); cerr != nil {
		if err == nil {
			err = cerr
		} else {
			err = fmt.Errorf("failed to close <whatever> after previous error (%w): %s", err, cerr)
		}
	}
}()
```
That way we can simply use the existing `err` to return any error that might occur during `Close()`, but it never overwrites an error that may have been thrown before.